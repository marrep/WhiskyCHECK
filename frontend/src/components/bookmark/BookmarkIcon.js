import BookmarkOff from "../../assets/icons/bookmarkOff.svg";
import BookmarkOn from "../../assets/icons/bookmarkOn.svg";
import styled from "styled-components";
import useBookmarks from "../../hooks/useBookmarks";

export default function BookmarkIcon({ id }) {
  const { toggleBookmark, bookmarks } = useBookmarks();

  return (
    <BookmarkIconWrapper>
      <BookmarkIconImage
        src={
          bookmarks.some((product) => product.id === id)
            ? BookmarkOn
            : BookmarkOff
        }
        alt=""
        onClick={() => toggleBookmark(id)}
      />
      <BookmarkIconText>Merken</BookmarkIconText>
    </BookmarkIconWrapper>
  );
}

export const BookmarkIconWrapper = styled.div`
  width: auto;
  text-decoration: none;
  overflow: hidden;
  justify-content: center;
  flex-direction: column;
  display: flex;
  align-items: center;
`;
export const BookmarkIconImage = styled.img`
  width: 80%auto;
`;
export const BookmarkIconText = styled.span`
  text-decoration: none;
  margin-top: 5px;
  font-weight: 200;
  font-size: 12px;
  display: block;
  color: #575757;
`;
