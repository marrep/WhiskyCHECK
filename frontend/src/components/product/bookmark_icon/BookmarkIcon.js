import BookmarkOff from "../../../assets/bookmarkOff.svg";
import BookmarkOn from "../../../assets/bookmarkOn.svg";

export default function BookmarkIcon({ saveThisItem, id, bookmarks }) {
  return (
    <div>
      <img
        src={
          bookmarks.some((product) => product.id === id)
            ? BookmarkOn
            : BookmarkOff
        }
        alt=""
        onClick={() => saveThisItem(id)}
      />
      <p>Merken</p>
    </div>
  );
}
