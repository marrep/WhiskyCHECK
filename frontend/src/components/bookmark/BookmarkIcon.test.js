import "jest-styled-components";
import BookmarkIcon from "./BookmarkIcon";
import { render } from "@testing-library/react";

const bookmarkMock = jest.fn();

describe("BookmarkIcon", () => {
  it("should render correctly", () => {
    const { container, getByText } = render(
      <BookmarkIcon
        id={1}
        toggleBookmark={bookmarkMock}
        bookmarks={[
          {
            image: "",
            id: 1,
          },
        ]}
      />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(getByText("Merken")).toBeTruthy();
  });
  it("should have a defined bookmark icon", () => {
    const { getByTestId } = render(
      <BookmarkIcon
        id={1}
        toggleBookmark={bookmarkMock}
        bookmarks={[
          {
            image: "",
            id: 1,
          },
        ]}
      />
    );
    const element = getByTestId("bookmarkIcon");
    expect(element).toBeDefined();
  });
});
