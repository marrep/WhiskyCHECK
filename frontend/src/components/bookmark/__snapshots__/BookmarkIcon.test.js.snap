// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookmarkIcon should render correctly 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  overflow: hidden;
  -webkit-text-decoration: none;
  text-decoration: none;
  width: auto;
}

.c1 {
  width: 80%auto;
}

.c2 {
  color: #575757;
  display: block;
  font-size: 12px;
  font-weight: 200;
  margin-top: 5px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  class="c0"
>
  <img
    alt=""
    class="c1"
    data-testid="bookmarkIcon"
    src="bookmarkOn.svg"
  />
  <span
    class="c2"
  >
    Merken
  </span>
</div>
`;
